packages/boot/test/bootstrapTests/benchmarkerator.md

```diff
@@ -0,0 +1,171 @@
+
+  An optional async method to perform any post-run teardown that you need to do.
+
+`rounds?: number`
+
+  The number of benchmark rounds that will be run if not overridden on the
+  command line.  If omitted, it defaults to `1`.
+
+#### The `BenchmarkContext` object
+
+The first parameter to each of the benchmark methods is a context object that
+provides various information about the execution.  It has the properties:
+
+`options: Record<string, string>`
+
+  Named options from the command line (see Command Line below).
+
+`argv: string[]`
+
+  Remaining unparsed arguments from the command line.
+
+`actors: Record<string, SmartWalletDriver>`
+
+  Wallet drivers for the various personae that exist in the test environment.
+  Currently present are `gov1`, `gov2`, and `gov3` (the governance committee);
+  `atom1` and `atom2` (ersatz ATOM-USD price feed oracles); and `alice`, `bob`,
+  and `carol` (arbitrary users to participate in interactions being exercised).
+
+`label: string`
+
+  The benchmark label, from the `addBenchmark` call that defined this benchmark.
+
+`rounds: number`
+
+  The total number of benchmark rounds that are to be executed in this run.
+
+`config?: Record<string, unknown>`
+
+  The configuration object that was returned by the `setup` method, if one was.
+
+## Executing a benchmark
+
+The benchmark that you define by incorporating The Benchmarkerator is a
+standalone Node executable.  You run it with the command:
+
+`node benchmark-yourbenchmarkname.js [OPTIONS...]`
+
+The supported command line options are:
+
+| Option | What |
+|--------|------|
+| `-r N`<br/>`--rounds N` | Execute _N_ rounds of each benchmark |
+| `-b PATT`<br/>`--benchmark PATT` | Only execute benchmarks matching _PATT_ (may be specified more than once; this is similar to Ava's `-m` option)|
+| `-o NAME VAL`<br/>`--option NAME VAL` | Set option _NAME_ to _VAL_ in the `context.options` record (may be specified more than once)  |
+| `-v`<br/>`--verbose` | Output verbose debug log messages as it runs |
+| `-l`<br/>`--local` | Use the `'local'` vat manager (instead of `'xs-worker'`; yields less realistic perf numbers but runs way faster and is much easier to debug) |
+| `-d`<br/>`--dump` | Output JSON-formated benchmark data to a file |
+| `-h`<br/>`--help` | Output this helpful usage information and then exit |
+
+additional unparsed _OPTIONS_ are passed to the benchmark itself in the `context.argv`
+array.
+
+## Results output
+
+Timing results and other collected metrics are output to _stdout_.  Two batches
+of information are provided: one for the setup phase and one for the benchmark
+rounds themselves (in aggregate).
+
+In addition, if you specify the `--dump` command line option, a JSON-formatted
+(i.e., machine readable) version of this same data will be output to the file
+`benchmark-NAME.json` in the current working directory (where _NAME_ is the name
+that you provided as the argument to `bench.run`).
```

Comment:
We can't control output generated by the benchmark itself (indeed, the current output even without the stats report and without the `--verbose` option is already incredibly verbose), so separating the benchmark's own output from the stats report output would require various subprocess voodoo that would impede debugging.

An alternative I _have_ been contemplating is changing the option to `--dump FILENAME`, i.e., specify the data destination directly on the command line rather than inferring the filename from the benchmark definition.  The thing deterring me from that path is that it would make it trickier to drive multiple benchmark files in a benchmarks directory from `yarn`, as the yarn script would itself be responsible for synthesizing the filename, which could be sketchy.